### ISSUE TITLE
Allow background music to continue playing while recording

### ISSUE THREAD

**Issue by blurtime** [NextLevel/NextLevel#85]
I am working on a project which may require music. Consequently, I wanted to use the following to lines which have worked quite well for me before I used NextLevel: 

    AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryAmbient, error: nil)
    AVAudioSession.sharedInstance().setActive(true, error: nil)

An error always occurs (for both lines) stating "expected declaration". In addition to that, I am not quite sure where to place this code... Should I put them in my view controller or try to incorporate them into the library? I would really appreciate some help!

**Comment by blurtime** [comment]
Since there doesn't seem to be a proper solution to this issue I am going to add a snippet which actually allows background music. However, the music is always interrupted whenever the camera view controller is left. 

Inside `viewWillAppear`:
    
        let nextLevel = NextLevel.shared
        if nextLevel.authorizationStatus(forMediaType: AVMediaTypeVideo) == .authorized &&
            nextLevel.authorizationStatus(forMediaType: AVMediaTypeAudio) == .authorized {
            do {
                try nextLevel.start()
            } catch {
                // prompt that the video has not been saved
                let alertController = UIAlertController(title: "Camera failed!", message: "The camera failed to start. Please restart this tab.", preferredStyle: .alert)
                let okAction = UIAlertAction(title: "OK", style: .default, handler: nil)
                alertController.addAction(okAction)
                self.present(alertController, animated: true, completion: nil)
            }
        } else {
            nextLevel.requestAuthorization(forMediaType: AVMediaTypeVideo)
            nextLevel.requestAuthorization(forMediaType: AVMediaTypeAudio)
        }

**Comment by piemonte** [comment]
hey @blurtime, did you try setting this property to false:

```swift
    /// Indicates whether the capture session automatically changes settings in the app’s shared audio session. By default, is `true`.
    public var automaticallyConfiguresApplicationAudioSession: Bool
  ```

**Comment by blurtime** [comment]
Yes, I tried setting it to false before I created this issue. It really made me wonder that the music still shut down. I do that in the my rather "incomplete" (see comment) solution, too.

**Comment by piemonte** [comment]
@blurtime ah, apologies, i read your message too quickly. sorry you encountered this bug, i'll look into it when i get a chance.

**Comment by blurtime** [comment]
No problem, nevertheless it's an awesome library! Keep up the great work!

**Comment by alvesl** [comment]
Hey guys, any updates on this? Should it be working now? Thanks!

**Comment by royherma** [comment]
+1

**Comment by piemonte** [comment]
unfortunately, I don't have the time to focus on this at the moment but would be awesome to support. opening the issue so we have it.

**Comment by piemonte** [comment]
Closing as won't fix. Background music during recording is system behavior controlled by AVAudioSession, not a library 
  feature.
